optuna/samplers/nsgaii/_sampler.py

```diff
@@ -345,6 +345,14 @@ def _fast_non_dominated_sort(
         population: List[FrozenTrial],
         directions: List[optuna.study.StudyDirection],
     ) -> List[List[FrozenTrial]]:
+        if self._constraints_func is not None:
+            for _trial in population:
+                _constraints = _trial.system_attrs.get(_CONSTRAINTS_KEY)
+                if _constraints is None:
+                    continue
```

Comment:
[Note] Trial without constraint values are handled by `_constrained_dominates` as follows:

https://github.com/optuna/optuna/blob/43370184bd24d99f4ab11093ea36646d55178595/optuna/samplers/nsgaii/_sampler.py#L479-L483

So, we can simply `continue` the for-loop.

Reply:
[Note] `sample_relative` also uses `_constrained_dominates`, but we do not need to check there since it calls `_fast_non_dominated_sort` at the top of the method indirectly. More specifically, the relationship between `sample_relative` and `_fast_non_dominated_sort` is as follows:
`sample_relative` --> `_collect_parent_populatio` --> `_select_elite_population` (if needed) --> `_fast_non_dominated_sort`